src/main.cpp

```diff
@@ -5713,6 +5713,9 @@ bool SendMessages(CNode* pto)
             pto->vAddrToSend.clear();
             if (!vAddr.empty())
                 pto->PushMessage(NetMsgType::ADDR, vAddr);
+            // we only send the big addr message once
+            if (pto->vAddrToSend.capacity() > 40)
+                pto->vAddrToSend = {};
```

Comment:
I don't think this frees up space.

To do so, I think you need the amazing idiom `std::vector<CAdress>().swap(pto->vAddrToSend)`. Since c++11 you can also use `pto->vAddrToSend.shrink_to_fit()` as a non-binding request.